### PR TITLE
Add support for linux VMs

### LINKED ISSUE #32: Create Linux VM
Now that tart provides linux VMs, it would be nice to be able to create Linux VMs from scratch with packer.

### PR DESCRIPTION
**Pull request by mayeut**
Resolves #32

I'm new to packer, go & provisioning in general so things might be ill formed but the example works.

Used https://ubuntu.com/server/docs/install/autoinstall-quickstart as a starting point for auto install.

A few notes:
It assumes the install process will shutdown the VM rather than restart it at the end of installation: the sample ubuntu autoinstall will just loop into the installer if the system is restarted rather than shutdown.
The auto-resize in the guest OS is probably less generic than the macOS one so ~~a `resize_disk_command` was introduced.~~ fixing the disk size is left to the user with generic shell provisioning steps.
The `boot_command` is hijacked to be used for the VM creation/installation rather than the following run.

### CODE REVIEW

File: example/ubuntu-22.04-prepare.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+mkdir cidata
+touch cidata/meta-data
+cat << 'EOF' > cidata/user-data
+#cloud-config
+autoinstall:
+  version: 1
+  identity:
+    hostname: ubuntu-server
+    password: "$6$exDY1mhS4KUYCE/2$zmn9ToZwTKLhCw.b4/b.ZRTIZM30JZ4QrOQ2aOXJ8yk96xpcCof0kxKwuX1kqLG/ygbJ1f8wxED22bTL4F46P0"
```
Comment (fkorotkov) [inline example/ubuntu-22.04-prepare.sh:22]: What's this password? Should it be `ubuntu`?

**Review by fkorotkov — COMMENTED** [review]
Thank you for the PoC! I think we should leave resizing of the disk out of scope for this PR. There is some very specific logic for macOS, I think in a follow up PR we can create similar steps for Linux and depending on `imageConfig["os"]` add different steps.

File: example/ubuntu-22.04-prepare.sh (hunk 1 above)
Comment (mayeut) [reply example/ubuntu-22.04-prepare.sh:22]: > Should it be `ubuntu`?

Well it is:
```
echo 'ubuntu' |openssl passwd -6 -stdin -salt exDY1mhS4KUYCE/2
$6$exDY1mhS4KUYCE/2$zmn9ToZwTKLhCw.b4/b.ZRTIZM30JZ4QrOQ2aOXJ8yk96xpcCof0kxKwuX1kqLG/ygbJ1f8wxED22bTL4F46P0
```

**Comment by mayeut** [comment]
> I think we should leave resizing of the disk out of scope for this PR

Ok, shall I remove everything or shall I leave the OS detection and the following ?
```go
if isLinux {
	ui.Error("Automatic partition resizing not implemented, guest OS might not see the full disk capacity.")
	return multistep.ActionContinue
 }
```
The original step didn't have a check on ExitStatus so it will just print a `diskutil: command not found` message but it won't fail the build.

**Comment by edigaryev** [comment]
> Ok, shall I remove everything or shall I leave the OS detection and the following ?

It seems that we can offload most of the OS-specific logic from `stepResize` to a user-managed `.hcl` (which knows the OS by design) and have no OS detection logic at all.

But this is for a separate PR, as noted by @fkorotkov.

**Comment by mayeut** [comment]
When rebasing on main, I had to leave a check to avoid failure with the disk parsing introduced on macOS.
I removed the specific resize_disk_command.

File: builder/tart/step_disk_resize.go [hunk 2]
```diff
@@ -26,6 +27,24 @@ func (s *stepResize) Run(ctx context.Context, state multistep.StateBag) multiste
 		return multistep.ActionContinue
 	}
 
+	// determine OS
```
Comment (fkorotkov) [inline builder/tart/step_disk_resize.go:30]: I wonder if it's easier to read `config.json` of the VM? What do you think? I don't have strong opinion on that.

Comment (edigaryev) [reply builder/tart/step_disk_resize.go:30]: To me `uname(1)` approach looks way easier, plus we don't assume that Tart stores VMs at specific paths, in specific file names, with specific format.

Ideally we could just drop `step_disk_resize.go` and move all of the logic to `*.hcl`'s (as I've mentioned before in https://github.com/cirruslabs/packer-plugin-tart/pull/33#issuecomment-1300305178), but that's obviously out of scope for this PR.

Comment (fkorotkov) [reply builder/tart/step_disk_resize.go:30]: Let's stick with `uname` but the resizing logic IMO should live in the plugin. For example, users should not try to figure out which disk to resize. IMO that's too advanced.

**Review by fkorotkov — APPROVED** [review]

**Review by edigaryev — APPROVED** [review]

**Pull request merged**